test/unit/model/PermissionsServiceTest.php

```diff
@@ -251,6 +254,56 @@ public function testDoNotSaveAnythingWhenThereIsNothingToSave(): void
         $this->service->savePermissions(false, $resource, []);
     }
 
+    public function testDuplicatedAddPermissionsAreNotPersistedTwice(): void
+    {
+        $resource = $this->createMock(core_kernel_classes_Class::class);
+        $resource->method('getUri')->willReturn('res1');
+
+        // Mock a strategy returning a duplicated permission
+        //
+        $this->strategy
+             ->method('normalizeRequest')
+             ->willReturn([
+                'add' => [
+                    'uid1' => ['GRANT', 'READ', 'WRITE', 'GRANT']
+                ]
+            ]);
+
+        $this->strategy->method('getPermissionsToAdd')->willReturn([
```

Comment:
Does it mean we should have fixed this method to avoid duplications?

Reply:
We may do that; however, I would do it _in addition to_ this fix (and not _instead of_ it).

That is: Ideally, the strategy used would not return duplicate permissions; however, IMO, `PermissionsService` and/or `DataBaseAccess` should always guarantee we don't try to persist duplicates.

For example, it a client extension provides an implementation that does not check for duplicates and we use that instead of a strategy provided by taoDacSimple itself, the service should still ensure duplicates are removed before reaching the database.

Reply:
ok, got it. Lets go with your fix for now. It covers all cases then :) 

Thanks!